packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -461,7 +464,7 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 			blob,
 			status: PendingBlobStatus.OnlinePendingUpload,
 			handleP: new Deferred(),
-			uploadP: this.uploadBlob(localId, blob),
+			uploadP: this.uploadBlob(localId, blob, signal),
```

Comment:
how will this work if the signal is aborted after upload but before ack, or before attach?

Reply:
i would probably lift all the abort signal stuff to only operate on handleP, which is the promise we return to the consumer, and can be directly rejected on abort, then its just internal clean up, which likely can also be centralized, rather than per step e.g. just a single function that walks the state and clean it all up.